Cache/Requests.lean

```diff
@@ -129,9 +129,30 @@ def downloadFiles (hashMap : IO.HashMap) (forceDownload : Bool) (parallel : Bool
       IO.Process.exit 1
   else IO.println "No files to download"
 
+def checkForToolchainMismatch : IO Unit := do
+  let mathlibToolchainFile := IO.mathlibDepPath / "lean-toolchain"
+  let downstreamToolchain ← IO.FS.readFile "lean-toolchain"
+  let mathlibToolchain ← IO.FS.readFile mathlibToolchainFile
+  if !(mathlibToolchain.trim = downstreamToolchain.trim) then
+    IO.println "Dependency Mathlib uses a different lean-toolchain"
+    IO.println s!"  Project uses {downstreamToolchain.trim}"
+    IO.println s!"  Mathlib uses {mathlibToolchain.trim}"
+    IO.println "\nIt is recommended that your project's toolchain matches Mathlib's toolchain"
+    IO.println s!"This can be achieved by copying the contents of the file {mathlibToolchainFile}
+into the lean-toolchain file at the root directory of your project"
```

Comment:
```suggestion
    IO.println s!"This can be achieved by copying the contents of the file `{mathlibToolchainFile}`
into the `lean-toolchain` file at the root directory of your project"
```